### ISSUE TITLE
NER ml backend example breaks during learning process

### ISSUE THREAD

**Issue by wojnarabc** [heartexlabs/label-studio-ml-backend#118]
For starters apologies for throwing two issues in just one day. I've much appreciated working with transformer branch of your project last year without considerable problems. Move to core ml-backend branch in which all transformer backends were already incorporated since that time appears to be challenging. 
Currently i'm unable to finish any learning NER process with following error thrown each time:
 File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/exceptions.py", line 39, in exception_f
    return f(*args, **kwargs)
```

  File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/api.py", line 70, in _train
    job = _manager.train(annotations, project, label_config, **params)

  File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 667, in train
    cls.get_or_create(project, label_config, force_reload=True, train_output=train_output)

  File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 471, in get_or_create
    if not cls.has_active_model(project) or force_reload or (cls.get(project).model_version != version and version is not None):  # noqa

  File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 447, in has_active_model
    return cls._key(project) in cls._current_model

TypeError: argument of type 'ModelWrapper' is not iterable
```
Steps to recreate:
0. Set up NER backend with Label Studio following guidelines from the repo (ner folder requirements.txt on top of it)
1. Load Dataset with any text attribute inside and set it up as NER ml template. 
2. Annotate some examples and start the learning process (initiated from Label Studio frontend in my case)
It starts to learn but throws above error after some time. 

Found this resolution for similar error but unfortunately doesn't work in my case. 

I'm currently using Label Studio 1.4.1post1 and the most recent version of ML backend.

**Comment by wojnarabc** [comment]
Hello @KonstantinKorotaev, can see that it was already merged with master branch so i've updated this one on my end as well. 
After the update, i've recreated environment and created new backend initialization (removing previous backend folder) but i'm still getting the same error as before. 
I've tried two scenarios before starting ml learning process in label studio. 
1.  Loading and annotating some data from scratch 
2. Loading preannotated json dataset 

This is full error message i get:

```
Traceback (most recent call last):
  File "/Users/maternaw/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 56, in get_result
    job_result = self.get_result_from_job_id(model_version)
  File "/Users/maternaw/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 110, in get_result_from_job_id
    assert isinstance(result, dict)
AssertionError
Some weights of the model checkpoint at bert-base-uncased were not used when initializing BertForTokenClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.decoder.weight', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias']
- This IS expected if you are initializing BertForTokenClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing BertForTokenClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Some weights of BertForTokenClassification were not initialized from the model checkpoint at bert-base-uncased and are newly initialized: ['classifier.weight', 'classifier.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
Process Process-1:
Traceback (most recent call last):
  File "/Applications/anaconda3/envs/ls_ml_backend/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/Applications/anaconda3/envs/ls_ml_backend/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 83, in job
    result = model.process_event(event, data, job_id, additional_params)
  File "/Users/user/Projects/label-studio-ml-backend/label_studio_ml/model.py", line 314, in process_event
    train_output = self.fit((), event=event, data=data, job_id=job_id, **additional_params)
  File "/Users/user/Projects/label-studio-ml-backend/my_ml_backend4/ner.py", line 507, in fit
    train_loader = DataLoader(
  File "/Applications/anaconda3/envs/ls_ml_backend/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 213, in __init__
    sampler = RandomSampler(dataset)
  File "/Applications/anaconda3/envs/ls_ml_backend/lib/python3.8/site-packages/torch/utils/data/sampler.py", line 93, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

```

**Comment by wojnarabc** [comment]
... when switched to LS 1.1.1 (used initially) second scenario started to work now. Had to create separate environment for that as there were some library clashes with the most current version of ml-backend. Still, doesn't work with 1.4.1post1 of LS.

**Comment by hemangjoshi37a** [comment]
You need to compare that the two are equal, not that one is in the other.

in this file : ` /home/<YOUR_USER_NAME>/anaconda3/lib/python3.8/site-packages/label_studio_ml/model.py`

replace this line : `return cls._key(project) in cls._current_model`
to this line : 
```
are_they_equal = cls._key(project) == cls._current_model
return are_they_equal
```
https://hjlabs.in

**Comment by lodeagcf** [comment]
Also seeing this `ValueError` with Bert backend.

**Comment by KonstantinKorotaev** [comment]
> ValueError: num_samples should be a positive integer value, but got num_samples=0

Please check that you are getting event with data.
You can check it in Webhook settings of your project.

**Comment by ritikp** [comment]
Hi @wojnarabc did you find the solution of above issue, if so please help me I am also facing same issue
